src/main/osd/osd_elements.c

```diff
@@ -156,6 +157,40 @@ static uint32_t blinkBits[(OSD_ITEM_COUNT + 31) / 32];
 #define IS_BLINK(item) (blinkBits[(item) / 32] & (1 << ((item) % 32)))
 #define BLINK(item) (IS_BLINK(item) && blinkState)
 
+#if defined(USE_ESC_SENSOR) || defined(USE_RPM_FILTER)
+
+typedef int (*getEscRpmOrFreqFnPtr)(int i);
+
+int getEscRpm(int i)
+{
+#ifdef USE_RPM_FILTER
+    if (motorConfig()->dev.useDshotTelemetry) {
+        return 100.0f / (motorConfig()->motorPoleCount / 2.0f) * getDshotTelemetry(i);
+    }
+#endif 
+    if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
+        return calcEscRpm(getEscSensorData(i)->rpm);
+    } else { 
+        return 0;
+    }
+}
+
+int getEscRpmFreq(int i) 
+{
+    return getEscRpm(i) / 60;
+}
+
+void renderOsdEscRpmOrFreq(getEscRpmOrFreqFnPtr escFnPtr, osdElementParms_t *element)
+{
+    int x = element->elemPosX;
+    int y = element->elemPosY;
+    for (int i=0; i < getMotorCount(); i++) {
+        char rpmStr[5];
+        tfp_sprintf(rpmStr, "%d", (*escFnPtr)(i));
```

Comment:
It looks like this will overrun the buffer. Need to allow for 5 characters for the max RPM and the trailing null.

Reply:
It is fixed. Good pickup. I have put a cap  of 99999 on the rpm value to avoid going over 6 digits to avoid overflow.  I have explicitly appended \0 to the end.